src/System.Collections.Concurrent/tests/ProducerConsumerCollectionTests.cs

```diff
@@ -761,8 +768,11 @@ public void ManyConcurrentAddsTakes_CollectionRemainsConsistent(int operations)
                 // avoid 0 as default(T), to detect accidentally reading a default value
                 for (int i = 1; i < (1 + operations); i++)
                 {
-                    Assert.True(c.TryAdd(i));
-                    added.Add(i);
+                    if (c.Count < MaxConcurrentCollectionSize)
+                    {
+                        Assert.True(c.TryAdd(i));
+                        added.Add(i);
+                    }
```

Comment:
> Well, unless the Take fails.

As long as each thread adds before it takes and the collection being tested isn't bounded such that the add could fail, the take should never fail.